### DISCUSSION TITLE
rewriting user queries / allow specifying alternate delimiters

### DISCUSSION THREAD

**Discussion by bhuntpenn** [junegunn/fzf#2873] [Ideas]
Hi all, I program in the Elixir programming language. Elixir has a naming convention for modules, which I'll illustrate. 

Module name : `Users.Pipelines.ValidateUser.Context`
File name : `./lib/users/pipelines/validate_user/context.ex`

Now, if I launch `fzf -i` and search for `Users.Pipelines.ValidateUser.Context`, no results are shown. 
But, if I launch `fzf -i` and search for `Users Pipelines ValidateUser Context`, the following result is (correctly) shown:

```
 ./lib/users/pipelines/validate_user/context.ex
```

That's great; at least I know where to go from here. However, I don't know if either of the following is possible: 

1. `fzf` to ignore '.' characters or use them as stopwords.
2. A hook so that when the string `Users.Pipelines.ValidateUser.Context` is pasted into the `fzf` search, it is re-written to `Users Pipelines ValidateUser Context`.

From reading the docs, I don't _think_ either is possible, but perhaps someone with more experience could confirm this for me? 

Thanks,

Bryan

**Comment by tg0h** [comment]
as a workaround, could you just search for the filename and content, presumably, the string `Users.Pipelines.ValidateUser.Context` is in the file `users/pipelines/validate_user/context.ex` ...
<img width="1484" alt="image" src="https://user-images.githubusercontent.com/7954426/182615191-e91caa5f-483b-48c5-babd-f848711b2eb8.png">

sample code to search for filename and contents can be found here https://github.com/junegunn/fzf/wiki/Examples#searching-file-contents

fzf code from screenshot here (i bind this to a keyboard shortcut)
```
#!bin/zsh
# search for filename and file contents
# edits the file after picking the fzf entry
# uses ripgrep (rg) and bat

function fzf-filename-and-content(){
  # searches for the filename and its contents in <dir>
  # <dir> defaults to cwd if none provided

  local dir=$1

  local fzfEntry=$(rg --follow --line-number --with-filename . $dir --color=always --field-match-separator ' '\
    | fzf --preview "bat --color=always {1} --highlight-line {2}" \
    --preview-window ~8,+{2}-5)

  if [ -n "$fzfEntry" ]; then
    local file lineNumber
    read -r file lineNumber _ <<< $fzfEntry
    nvim $file +$lineNumber -c 'normal! zz'
  fi
}

```

**Reply by bhuntpenn** [reply]
That's pretty clever and an approach I didn't consider. I'll snarf that code if you don't mind and use it as an alternative approach. Thanks!
